src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs

```diff
@@ -0,0 +1,444 @@
+            [".pfm"] = new ContentTypeMapping("application/octet-stream", null, "*.pfm", 1),
+            [".prm"] = new ContentTypeMapping("application/octet-stream", null, "*.prm", 1),
+            [".prx"] = new ContentTypeMapping("application/octet-stream", null, "*.prx", 1),
+            [".psd"] = new ContentTypeMapping("application/octet-stream", null, "*.psd", 1),
+            [".psm"] = new ContentTypeMapping("application/octet-stream", null, "*.psm", 1),
+            [".psp"] = new ContentTypeMapping("application/octet-stream", null, "*.psp", 1),
+            [".qxd"] = new ContentTypeMapping("application/octet-stream", null, "*.qxd", 1),
+            [".rar"] = new ContentTypeMapping("application/octet-stream", null, "*.rar", 1),
+            [".sea"] = new ContentTypeMapping("application/octet-stream", null, "*.sea", 1),
+            [".smi"] = new ContentTypeMapping("application/octet-stream", null, "*.smi", 1),
+            [".snp"] = new ContentTypeMapping("application/octet-stream", null, "*.snp", 1),
+            [".thn"] = new ContentTypeMapping("application/octet-stream", null, "*.thn", 1),
+            [".toc"] = new ContentTypeMapping("application/octet-stream", null, "*.toc", 1),
+            [".u32"] = new ContentTypeMapping("application/octet-stream", null, "*.u32", 1),
+            [".xsn"] = new ContentTypeMapping("application/octet-stream", null, "*.xsn", 1),
+            [".xtp"] = new ContentTypeMapping("application/octet-stream", null, "*.xtp", 1),
+        };
+
+    internal ContentTypeMapping ResolveContentTypeMapping(string relativePath, TaskLoggingHelper log)
+    {
+        var builtIn = ResolveBuiltIn(relativePath, log);
+        foreach (var mapping in customMappings)
+        {
+            if (mapping.Matches(Path.GetFileName(relativePath)))
+            {
+                // If a custom mapping matches, it wins over the built-in
+                log.LogMessage(MessageImportance.Low, $"Matched {relativePath} to {mapping.MimeType} using pattern {mapping.Pattern}");
+                return mapping;
+            }
+            else
+            {
+                log.LogMessage(MessageImportance.Low, $"No match for {relativePath} using pattern {mapping.Pattern}");
+            }
+        }
+
+        return builtIn;
+    }
+
+    private ContentTypeMapping ResolveBuiltIn(string relativePath, TaskLoggingHelper log)
+    {
+        var extension = Path.GetExtension(relativePath);
+        if (extension == ".gz" || extension == ".br")
+        {
+            var fileName = Path.GetFileNameWithoutExtension(relativePath);
+            if (Path.GetExtension(fileName) != "")
+            {
+                var result = ResolveBuiltIn(fileName, log);
+                // If we don't have a specific mapping for the other extension, use any mapping available for `.gz` or `.br`
+                return result.Equals(default) && _builtInMappings.TryGetValue(extension, out var compressed) ?
```

Comment:
`ContentTypeMapping` doesn't override `Equals()`, so `result.Equals(default)` will always return `false`, right?

Reply:
Maybe `ResolveBuiltIn()` should return a `ContentTypeMapping?`.

Reply:
I've changed the comparison